cli/flox/src/commands/edit.rs

```diff
@@ -443,6 +473,297 @@ mod tests {
         Edit::make_interactively_recoverable(result).expect_err("should return unhandled Err");
     }
 
+    /// Error due to empty vars and no editor in PATH
+    #[test]
+    fn test_determine_editor_from_vars_not_found() {
+        let visual_var = "".to_owned();
+        let editor_var = "".to_owned();
+
+        let tmp1 = tempdir().expect("should create tempdir");
+        let tmp2 = tempdir().expect("should create tempdir");
+        let tmp3 = tempdir().expect("should create tempdir");
+
+        let path_var = std::env::join_paths([&tmp1, &tmp2, &tmp3].map(|d| d.path()))
+            .expect("should path-join tmpdirs")
+            .into_string()
+            .expect("should convert paths from OsString to String");
+
+        Edit::determine_editor_from_vars(visual_var, editor_var, path_var)
+            .expect_err("should error with editor not found");
+
+        assert!(tmp1.path().is_dir());
+        assert!(tmp2.path().is_dir());
+        assert!(tmp3.path().is_dir());
```

Comment:
non-blocking: I dont think we need to assert that our tempdirs are dirs, I'd trust the tempdir function to be correct here..

Reply:
commented that this prevents early drops via passing ownership to other funcs while also declaring what the depended-upon side-effect on the fs is.